src/main/java/de/tum/in/www1/artemis/service/TextSubmissionService.java

```diff
@@ -129,7 +129,7 @@ else if (textExercise.isExamExercise()) {
      */
     public Optional<TextSubmission> getRandomTextSubmissionEligibleForNewAssessment(TextExercise textExercise, boolean skipAssessmentQueue, boolean examMode, int correctionRound) {
         // If automatic assessment is enabled and available, try to learn the most possible amount during the first correction round
-        if (textExercise.isFeedbackSuggestionsEnabled() && athenaSubmissionSelectionService.isPresent() && !skipAssessmentQueue && correctionRound == 0) {
+        if (textExercise.getFeedbackSuggestionsEnabled() && athenaSubmissionSelectionService.isPresent() && !skipAssessmentQueue && correctionRound == 0) {
```

Comment:
Why isn't there a `ProgrammingSubmissionService` with `getRandomProgrammingSubmissionEligibleForNewAssessment`, am I missing something?

Reply:
I completely forgot that. I added this now.